bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveSwitchAllCommandClass.java

```diff
@@ -0,0 +1,207 @@
+                break;
+            }
+            case SWITCH_ALL_GET: {
+                logger.warn("Command {} not implemented.", (Object)command);
+                return;
+            }
+            case SWITCH_ALL_REPORT: {
+                logger.trace("Process Switch All Report");
+                this.processSwitchAllReport(serialMessage, offset, endpoint);
+                break;
+            }
+            default: {
+                logger.warn(String.format("Unsupported Command 0x%02X for command class %s (0x%02X).", command, this.getCommandClass().getLabel(), this.getCommandClass().getKey()));
+            }
+        }
+    }
+
+    protected void processSwitchAllReport(SerialMessage serialMessage, int offset, int endpoint) {
+        int mode = serialMessage.getMessagePayloadByte(offset + 1);
+        switch (mode) {
+            case SWITCH_ALL_EXCLUDED:
+                logger.debug("NODE {}: Switch All report, device is not included in either All On or All Off groups.", (Object)this.getNode().getNodeId());
+                break;
+            case SWITCH_ALL_INCLUDE_ON_ONLY:
+                logger.debug("NODE {}: Switch All report, device included in All On group.", (Object)this.getNode().getNodeId());
+                break;
+            case SWITCH_ALL_INCLUDE_OFF_ONLY:
+                logger.debug("NODE {}: Switch All report, device included in All Off group.", (Object)this.getNode().getNodeId());
+                break;
+            case SWITCH_ALL_INCLUDE_ON_OFF:
+                logger.debug("NODE {}: Switch All report, device included in All On and All Off group.", (Object)this.getNode().getNodeId());
+                break;
+            default:
+                logger.warn(String.format("Unsupported Switch All mode 0x%02X.", mode));
+        }
+        ZWaveCommandClassValueEvent zEvent = new ZWaveCommandClassValueEvent(this.getNode().getNodeId(), endpoint, this.getCommandClass(), mode);
+        this.getController().notifyEventListeners(zEvent);
+    }
+
+    public SerialMessage getValueMessage() {
+        logger.debug("NODE {}: Creating new message for command SWITCH_ALL_GET", (Object)this.getNode().getNodeId());
+        SerialMessage result = new SerialMessage(this.getNode().getNodeId(), SerialMessage.SerialMessageClass.SendData, SerialMessage.SerialMessageType.Request, SerialMessage.SerialMessageClass.ApplicationCommandHandler, SerialMessage.SerialMessagePriority.Get);
+        byte[] newPayload = { 
+        		(byte)this.getNode().getNodeId(), 
+        		2, 
+        		(byte) this.getCommandClass().getKey(), 
+        		(byte) SWITCH_ALL_GET };
+        result.setMessagePayload(newPayload);
+        return result;
+    }
+
+    /**
+     * Create a new SwitchAll set message
+     * @param mode as (0x00 - Exclude, 0x01 Only All On, 0x02 Only All Off, 0xFF Both All on and All off)
+     * @return SerialMessage
+     */
+    public SerialMessage setValueMessage(int mode) {
+        if (!this.isGetSupported) {
```

Comment:
I'm not sure what you mean? This is the SET command - why does it have the ```isGetSupported``` flag? This should be removed.